.github/workflows/publish-github-release.yml

```diff
@@ -0,0 +1,215 @@
+name: Publish GitHub Release as newest
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 3 * * 1-5'
+
+jobs:
+  publish-release:
+    name: Publish GitHub Release
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: write
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+
+      - name: Get latest draft name
+        id: get_latest_draft
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          latest_release=$(gh release list --json name,tagName,isDraft,isPrerelease -L 1 --jq '.[0]')
+          is_latest_draft=$(jq -r '.isDraft' <<< $latest_release)
+          echo "is_latest_draft=$is_latest_draft" >> $GITHUB_OUTPUT
+
+          if [ "$is_latest_draft" != "true" ]; then
+              echo "No draft found"
+              exit 0
+          fi
+
+          latest_draft_version=$(jq -r '.tagName' <<< $latest_release)
+          echo "latest_draft_version=$latest_draft_version" >> $GITHUB_OUTPUT
+
+          latest_draft_name=$(jq -r '.name' <<< $latest_release)
+          echo "latest_draft_name=$latest_draft_name" >> $GITHUB_OUTPUT
+
+      - name: Get previous run ID
+        id: get_previous_run
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          workflow_name="$(basename "$GITHUB_WORKFLOW")"
+
+          # Get the last successful scheduled run ID
+          previous_run_id=$(gh run list --workflow=$workflow_name --event=schedule --status=success --limit 1 --json databaseId --jq '.[0].databaseId // empty')
+
+          if [ -n "$previous_run_id" ] && [ "$previous_run_id" != "null" ]; then
+            echo "Found previous successful scheduled run: $previous_run_id"
+            echo "previous_run_id=$previous_run_id" >> $GITHUB_OUTPUT
+            echo "has_previous_run=true" >> $GITHUB_OUTPUT
+          else
+            echo "No previous successful scheduled run found"
+            echo "has_previous_run=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Download previous run state
+        id: previous_state
+        if: steps.get_previous_run.outputs.has_previous_run == 'true'
+        continue-on-error: true
+        uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093 # v4.3.0
+        with:
+          name: release-info
+          run-id: ${{ steps.get_previous_run.outputs.previous_run_id }}
+
+      - name: Parse previous state
+        id: parse_previous_state
+        if: steps.get_previous_run.outputs.has_previous_run == 'true'
+        run: |
+          if [ -f "release-info.json" ]; then
+            previous_release_tag=$(jq -r '.release_tag // empty' release-info.json)
```

Comment:
⛏️ You can probably use `fromJSON(...) >> GITHUB_OUTPUT`  and skip manually parsing each field.

Reply:
Can't do that. I could use the `fromJSON()` function here, but differently than your suggestion.

Use `fromJSON` instead of `jq`:
```
- name: Read JSON configuration
  id: json-reader
  run: |
    content=$(cat release-info.json)
    echo "release-info-json=$content" >> $GITHUB_OUTPUT
    
- name: Process JSON data
  run: |
    echo "Release tag: ${{ fromJSON(steps.json-reader.outputs.release-info-json).release_tag }}"
    echo "Initial state: ${{ fromJSON(steps.json-reader.outputs.release-info-json).initial_state }}"
    echo "Changed to: ${{ fromJSON(steps.json-reader.outputs.release-info-json).changed_to_state }}"   
```

I'd rather have the fields parsed as outputs to use them directly, without needing to do anything with them in the steps that use values.